RS-Clone

Gregory-Vinokur opened the pull request:
1. Task: https://github.com/rolling-scopes-school/tasks/blob/master/tasks/rsclone/rsclone.md
2. Screenshot: [![VK-Clone.jpg](https://i.postimg.cc/4xFMV8mz/VK-Clone.jpg)](https://postimg.cc/k2KfdvN4)
3. Deploy: [vk-clone-rs-clone-party-team.netlify.app](https://vk-clone-rs-clone-party-team.netlify.app/)
4. Done 20.02.2023 / deadline 21.02.2023
5. Score 620 / 620

6. Features:

**Общее**
1. Приложение переведено на 2 языка (рус, анг) – 10 б.
2. Реализована темная тема – 10 б.
3. Использован MVC паттерн – 10 б.
4. Роутинг на нативном TS – 5 б.
5. Кастомная страница 404 – 5 б.
6. Адаптив всех страниц приложения – 10 б.

**1. Раздел “Моя страница” – всего: 120 баллов**
- Загрузка обложки страницы – 10 б.
- Загрузка аватара – 10 б.
- Прогресс-бар загрузки аватара и обложки страницы (появляется в момент смены аватара или обложки под статусом пользователя) – 5 б.
- Смена имени  пользователя – 5 б.
- Смена статуса пользователя – 5 б.
- Возможность создания текстовых постов (все посты будут появляться в общей ленте, пока не будут удалены) – 5 б.
- Репост созданного поста со страницы другого пользователя – 10 б.
- Создание постов с картинкой (нажать на кнопку камеры, выбрать фото и дождаться загрузки) – 10 б.
- Созданный пост или репост можно удалить, при этом изменятся счетчики репостов у соответствующего сообщества или пользователя со страницы которого сделан репост – 10 б.
- Присутствует анимация загрузки картинки поста (при создании новости и добавлении фото)– 5 б.
- Если стена пользователя пустая – выводится соответствующее сообщение – 5 б.
- Посещение страниц других пользователей (из раздела подписки или рекомендованные пользователи) – 10 б.
- Раздел рекомендованные пользователи (рекомендации сделаны на основании подписок, иногда случайных пользователей) – 10 б.
- Отображение избранной музыки на личной странице – 10 б.
- Информация храниться в БД и загружается при каждом посещении страницы – 10 б.

**2. Раздел “Музыка” – всего 80 баллов**
- Музыка обновляется каждый день и содержит топ-20 треков мирового чарта – 5 б.
- Главный плеер (самый верхний) позволяет перелистывать музыку, управлять паузой и воспроизведением, отражает прогресс-бар и текущую громкость – 15 б.
- Треки можно добавить в избранное, избранный трек отмечен, сохраняется в разделе “Моя музыка” и отображается на личной странице – 10 б.
- Реализован раздел “Моя музыка”, который содержит коллекцию избранных треков – 10 б.
- Избранные треки сохраняются в БД и загружаются при каждой авторизации и после перезагрузки страницы – 10 б.
- Текущий проигрываемый трек выделен анимацией – 5 б.
- Присутствует поле для поиска музыки, музыка ищется по исполнителям, названиям песен, в поиске как правило выводятся актуальные коллекции – 20 б.
- Если треков не найдено, выводится соответствующее сообщение – 5 б.

**3. Раздел "Сообщения"  – всего 180 баллов.**
- Отправка сообщений в общий чат - 10б.
- Сортировка и лимит отображаемых сообщений в окне чата - 10б.
- Возможность удалить собственное сообщение(в общем чате, в личных сообщениях и в группах) - 15б.
- Возможность редактировать собственное сообщение(в общем чате, в личных сообщениях и в группах) - 15б.
- Визуальное выделение сообщений пользователя (собственные сообщения отображаются в правой части окна чата, сообщения других пользователей в левой) - 10б.
- Модальное окно с возможностью подписаться/отписаться на пользователя и/или написать ему сообщение - 10б.
- Поиск пользователя по имени - 10б.
- Отправка сообщений конкретному пользователю - 10б.
- Создание групп - 10б.
- Поиск существующих групп по названию - 10б.
- Если при поиске групп или пользователей ничего не найдено выводится соответствующее сообщение - 5б.
- При отсутствии сообщений в чате выводится соответствующая надпись - 5б.
- Выбранные вкладки(чат, группы и личные сообщения), а также выбранные пользователи или группы выделяются стилями - 5б.
- Отправка сообщений в группу - 10б.
- Возможность войти или покинуть группу - 10б.
- Сообщения написанные другими пользователями подгружаются автоматически(в общем чате, в личных сообщениях и в группах) - 20б.
- Оповещение о новых сообщениях. При получении нового сообщения появляется иконка рядом с именем пользователя или названием группы(если чат не открыт), при открытии чата иконка пропадает. Так же иконка появляется при загрузке приложения, если есть непрочитанные сообщения(в личных сообщениях или в группах) - 15б.

**4. “Авторизация” – всего 55 баллов**
- При первом запуске приложения отображается страница с кнопками "Вход" и "Авторизация" - 5б.
- При нажатии на регистрацию пользователь вводит свои почту, имя и пароль. Ввод проверяется на валидность. При успешном вводе данные отправляются на сервер.  - 20б.
- При нажатии на вход пользователь вводит свою почту и пароль, которые проверяется на валидность, и, в случае существующего аккаунта, успешно входит в приложение. При успешном вводе данные отправляются на сервер. - 20б.
- При повторном запуске приложения, в случае если пользователь авторизован, появляется предложение продолжить работу либо сменить пользователя. -5б.
- В случае, если пользователь забыл пароль, есть возможность сброса пароля через отправку письма по электронной почте. - 5б.

**5. Раздел “Новости” – всего 70 баллов**
- Отрисовка всех постов пользователей и сообществ из базы данных. -10б.
- Возможность лайков постов с счётчиком и запоминанием количества лайков после перезагрузки страницы. -10б.
- Возможность репостов на личную страницу пользователя постов с счётчиком и запоминанием количества репостов после перезагрузки страницы. -10б.
- Возможность комментировать посты. -5б.
- Комментарии под постами сохраняются. -5б.
- Комментарии под постами можно лайкать с сохранение количества лайков после перезагрузки страницы. -5б.
- Комментарии под постами можно редактировать и удалять с сохранением состояния после перезагрузки. -5б.
- Реализована бесконечная лента с throttle функцией: сначала загружаются 10 постов. Когда пользователь дойдёт до конца страницы - загружаются ещё 10 постов и т.д. -10б. 
- Возможность перехода на страницу сообщества при нажатии на лого сообщества. -5б.
- Отображение времени, прошедшего с даты публикации поста. -5б.

**6. Раздел “Сообщества” – всего 65 баллов**
- Отображение всех сообществ соц. сети с названием тематики и количеством подписчиков из базы данных. -10 б.
- Переход на страницу каждого сообщества. - 10б.
- Возможность подписаться/отписаться. - 10б.
- Отрисовка подписчиков и их количества. -5б.
- Возможность перейти на личную страницу подписчика по нажатию на его аватар. -5б.
- Генерация постов с использованием стороннего API согласно тематике сообщества. -20б.
- Отображение времени с момента генерации пота. -5б.

Gregory-Vinokur committed:
init: project setup

Gregory-Vinokur committed:
feat: add basic layout

Gregory-Vinokur committed:
feat: add authorization via Firebase

Gregory-Vinokur committed:
feat: add eslint and prettier configs

Gregory-Vinokur committed:
feat: add navbar and route to profilePage

Gregory-Vinokur committed:
Merge pull request #1 from Gregory-Vinokur/feat/addAuth

Feat/add auth

artemkamyshenkov committed:
fead: add upload ava and cover for profile user

artemkamyshenkov committed:
feat: start upload avatar

M0rl0ck committed:
feat: add chat

artemkamyshenkov committed:
resolve conflict

artemkamyshenkov committed:
Merge pull request #2 from Gregory-Vinokur/my-page

My page

M0rl0ck committed:
Merge branch 'develop' into messages

M0rl0ck committed:
Merge pull request #3 from Gregory-Vinokur/messages

feat: add chat

M0rl0ck committed:
feat: add scroll messages to end

Gregory-Vinokur committed:
feat: finish auth logic

Gregory-Vinokur committed:
Merge branch 'develop' into feat/AuthLogic

Gregory-Vinokur committed:
Merge pull request #4 from Gregory-Vinokur/feat/AuthLogic

feat: finish auth logic

M0rl0ck committed:
feat: add delete message

M0rl0ck committed:
Merge branch 'develop' into messages

M0rl0ck committed:
feat: add limit messages

M0rl0ck committed:
feat: add sort messages

artemkamyshenkov committed:
feat: add synchronization for user and message

Gregory-Vinokur committed:
feat: add show/hide button for password and Logout button

Gregory-Vinokur committed:
Merge pull request #6 from Gregory-Vinokur/feat/AuthLogic

feat: add show/hide button for password and Logout button

M0rl0ck committed:
Merge pull request #5 from Gregory-Vinokur/messages

Messages

Gregory-Vinokur committed:
fix: fix hide/show password button layout

M0rl0ck committed:
fix: fix style

artemkamyshenkov committed:
resolve conflict merge

artemkamyshenkov committed:
Merge pull request #8 from Gregory-Vinokur/my-page

feat: add synchronization for user and message

M0rl0ck committed:
feat: add translation

M0rl0ck committed:
Merge branch 'develop' into messages

M0rl0ck committed:
Merge branch 'messages' into lang

M0rl0ck committed:
refactor: resolve conflict

M0rl0ck committed:
Merge pull request #7 from Gregory-Vinokur/messages

fix: fix style

M0rl0ck committed:
refactor: edit routing

M0rl0ck committed:
refactor: refactor mvc

artemkamyshenkov committed:
feat: add created user post

M0rl0ck committed:
Merge pull request #10 from Gregory-Vinokur/messages

Messages

artemkamyshenkov committed:
resolve conflict

Gregory-Vinokur committed:
feat: add cats posts to NewsPage

Gregory-Vinokur committed:
feat: merge addNews

Gregory-Vinokur committed:
Merge pull request #11 from Gregory-Vinokur/feat/addNews

Feat/add news

artemkamyshenkov committed:
feat: add new user

artemkamyshenkov committed:
Merge branch 'develop' into my-page

M0rl0ck committed:
refactor: replaced input with textarea

M0rl0ck committed:
Merge branch 'develop' into messages

artemkamyshenkov committed:
feat: add create news

artemkamyshenkov committed:
fix: bug with created post

artemkamyshenkov committed:
feat: add func delete post

artemkamyshenkov committed:
feat: add load foto in user post

artemkamyshenkov committed:
feat: add new function in user profile

artemkamyshenkov committed:
Merge pull request #12 from Gregory-Vinokur/my-page

My page

artemkamyshenkov committed:
add todo

artemkamyshenkov committed:
Merge branch 'develop' into my-page

M0rl0ck committed:
feat: add subscript to user

M0rl0ck committed:
Merge branch 'develop' into messages

M0rl0ck committed:
refactor: refactor style

M0rl0ck committed:
Merge pull request #13 from Gregory-Vinokur/messages

Messages

artemkamyshenkov committed:
Merge branch 'develop' into my-page

artemkamyshenkov committed:
feat: add render user friends

artemkamyshenkov committed:
Merge pull request #14 from Gregory-Vinokur/my-page

Add render user friends and open friend in new tab

M0rl0ck committed:
feat: add chat rooms page

M0rl0ck committed:
Merge branch 'develop' into messages

Gregory-Vinokur committed:
feat: add infinity scroll to NewsPage

Gregory-Vinokur committed:
Merge pull request #15 from Gregory-Vinokur/feat/addNews

feat: add infinity scroll to NewsPage

M0rl0ck committed:
Merge branch 'develop' into messages

artemkamyshenkov committed:
feat: add recommended friends

artemkamyshenkov committed:
Merge pull request #16 from Gregory-Vinokur/my-page

feat: add recommended friends

M0rl0ck committed:
feat: add dialog

M0rl0ck committed:
Merge branch 'develop' into messages

M0rl0ck committed:
Merge pull request #17 from Gregory-Vinokur/messages

Messages

M0rl0ck committed:
fix: fix link: write user

M0rl0ck committed:
fix: fix scroll messages

M0rl0ck committed:
feat: add delete message

artemkamyshenkov committed:
feat: add progress bar upload img in user page

feat: add render empty block

feat: add eng lang for user page

M0rl0ck committed:
feat: add modal user window

artemkamyshenkov committed:
Merge pull request #18 from Gregory-Vinokur/my-page

feat: add progress bar upload img in user page

artemkamyshenkov committed:
fix: fix lang change

artemkamyshenkov committed:
Merge pull request #19 from Gregory-Vinokur/my-page

fix: fix lang change

ardashevdanil reviewed (approved):
В целом ок. Из минусов - методы модели вызываются во вьюхах\компонентах

M0rl0ck committed:
feat: add icon new message

M0rl0ck committed:
Merge branch 'develop' into messages

M0rl0ck committed:
fix: fix create rooms

M0rl0ck committed:
refactor: disabled limit

M0rl0ck committed:
Merge pull request #20 from Gregory-Vinokur/messages

Messages

M0rl0ck committed:
feat: add find user

M0rl0ck committed:
refactor: replace method debounce with function

M0rl0ck committed:
refactor: edite style

M0rl0ck committed:
fix: fix input style

M0rl0ck committed:
Merge pull request #21 from Gregory-Vinokur/messages

Messages

artemkamyshenkov committed:
add music page

artemkamyshenkov committed:
Merge branch 'develop' into feature/music

artemkamyshenkov committed:
feat: add style for music page

Gregory-Vinokur committed:
feat: add comments section in posts

artemkamyshenkov committed:
Merge pull request #22 from Gregory-Vinokur/feature/music

Feature/music

Gregory-Vinokur committed:
Merge pull request #23 from Gregory-Vinokur/feat/addNews

feat: add comments section in posts

M0rl0ck committed:
feat: add avatar to modal window

artemkamyshenkov committed:
feat: add play/pause music

M0rl0ck committed:
fix: fix send message to undefind

artemkamyshenkov committed:
Merge pull request #24 from Gregory-Vinokur/feature/music

feat: add play/pause music

M0rl0ck committed:
feat: add create group

M0rl0ck committed:
Merge branch 'develop' into messages

artemkamyshenkov committed:
feat: add search music

M0rl0ck committed:
feat: add create group

artemkamyshenkov committed:
feat: add new style for music

artemkamyshenkov committed:
Merge pull request #25 from Gregory-Vinokur/feature/music

Feature/music

M0rl0ck committed:
feat: add send message to group

M0rl0ck committed:
Merge branch 'develop' into messages

artemkamyshenkov committed:
feat: add user favorite music in bd

M0rl0ck committed:
feat: add search group

M0rl0ck committed:
feat: add delete group message

artemkamyshenkov committed:
Merge pull request #26 from Gregory-Vinokur/feature/music

feat: add user favorite music in bd

M0rl0ck committed:
Merge branch 'develop' into messages

M0rl0ck committed:
Merge pull request #27 from Gregory-Vinokur/messages

Messages

artemkamyshenkov committed:
feat: add container for user music

artemkamyshenkov committed:
Merge branch 'develop' into feature/user-music

artemkamyshenkov committed:
feat: render favorite music on user page

artemkamyshenkov committed:
feat add view music in order page

artemkamyshenkov committed:
feat: add user post in newsline

artemkamyshenkov committed:
feat: add new input change name and status

artemkamyshenkov committed:
fix: bug with btn upload cover

Gregory-Vinokur committed:
feat: add CommunitiesPage, reposts and routing with query string

Gregory-Vinokur committed:
Merge branch 'develop' into feat/addCommunities

Gregory-Vinokur committed:
Merge pull request #28 from Gregory-Vinokur/feat/addCommunities

feat: add CommunitiesPage, reposts and routing with query string

M0rl0ck committed:
refactor: edit style

M0rl0ck committed:
Merge branch 'develop' into messages

M0rl0ck committed:
Merge pull request #29 from Gregory-Vinokur/messages

Messages

artemkamyshenkov committed:
feat: add user like

artemkamyshenkov committed:
Merge branch 'develop' into feature/user-music

artemkamyshenkov committed:
resolve conflict

artemkamyshenkov committed:
Merge pull request #30 from Gregory-Vinokur/feature/user-music

Feature/user music

artemkamyshenkov committed:
feat: add style for header

feat: add style for footer

feat: add style for messenger

feat: add style for groups

feat: add new icon

M0rl0ck committed:
fix: update user info

M0rl0ck committed:
refactor: add emit setUser

M0rl0ck committed:
Merge pull request #31 from Gregory-Vinokur/messages

fix: update user info

artemkamyshenkov committed:
Merge branch 'develop' into feature/styles

artemkamyshenkov committed:
Merge pull request #32 from Gregory-Vinokur/feature/styles

feat: add style for header

artemkamyshenkov committed:
feat: add upload ava on header

artemkamyshenkov committed:
Merge branch 'develop' into feature/styles

artemkamyshenkov committed:
Merge pull request #33 from Gregory-Vinokur/feature/styles

Add user ava on header

artemkamyshenkov committed:
feat: add dark theme

artemkamyshenkov committed:
Merge pull request #34 from Gregory-Vinokur/feature/dark-theme

feat: add dark theme

M0rl0ck committed:
feat: add adaptive style

M0rl0ck committed:
Merge branch 'develop' into messages

M0rl0ck committed:
feat: add burger menu

M0rl0ck committed:
fix: add viewport to html

M0rl0ck committed:
Merge pull request #35 from Gregory-Vinokur/messages

Messages

artemkamyshenkov committed:
feat: add routing for user profile

artemkamyshenkov committed:
fix: fix recommended users if user doesnt have any subscription

Gregory-Vinokur committed:
feat: add favicon, required nameInput for sign up and id for post container

artemkamyshenkov committed:
fix: fix subscribe user

Gregory-Vinokur committed:
fix: fix news

Gregory-Vinokur committed:
Merge pull request #36 from Gregory-Vinokur/fix/fix_reposts

feat: add favicon, required nameInput for sign up

artemkamyshenkov committed:
Merge pull request #37 from Gregory-Vinokur/feature/user-page

Feature/user page

artemkamyshenkov committed:
fix: fix ava header

Gregory-Vinokur committed:
feat: add id to reposts

Gregory-Vinokur committed:
feat: add id to Post Interface

Gregory-Vinokur committed:
Merge pull request #38 from Gregory-Vinokur/fix/fix_reposts

feat: add id to reposts

artemkamyshenkov committed:
Merge branch 'develop' into feature/user-page

Gregory-Vinokur committed:
fix: fix userPost id

M0rl0ck committed:
fix: fix update user

M0rl0ck committed:
fix: fix display time message

M0rl0ck committed:
Merge pull request #39 from Gregory-Vinokur/messages

Messages

artemkamyshenkov committed:
Merge branch 'develop' into feature/user-page

artemkamyshenkov committed:
feat: add delete repost

artemkamyshenkov committed:
Merge pull request #40 from Gregory-Vinokur/feature/user-page

Feature/user page

M0rl0ck committed:
feat: add adaptive style

M0rl0ck committed:
Merge branch 'develop' into messages

M0rl0ck committed:
Merge pull request #41 from Gregory-Vinokur/messages

Messages

artemkamyshenkov committed:
feat: add user repost

M0rl0ck committed:
feat: add edit message

M0rl0ck committed:
Merge pull request #42 from Gregory-Vinokur/messages

feat: add edite message

artemkamyshenkov committed:
Merge branch 'develop' into feature/user-page

artemkamyshenkov committed:
feat: add delete repost user

artemkamyshenkov committed:
Merge pull request #43 from Gregory-Vinokur/feature/user-page

Feature/user page

artemkamyshenkov committed:
fix: fix repost counter

artemkamyshenkov committed:
refactor view profile

artemkamyshenkov committed:
Merge pull request #44 from Gregory-Vinokur/feature/user-page

Feature/user page

artemkamyshenkov committed:
Merge branch 'main' into develop

artemkamyshenkov committed:
fix: fix search music style

artemkamyshenkov committed:
Merge pull request #45 from Gregory-Vinokur/fix/search-music

fix: fix search music style

M0rl0ck committed:
feat: add no Message title

M0rl0ck committed:
Merge pull request #46 from Gregory-Vinokur/messages

feat: add no Message title

artemkamyshenkov committed:
Update README.md

artemkamyshenkov committed:
Update README.md

Gregory-Vinokur committed:
feat: add adaptive and translate to communitiesPage, loginPage, newsPage and errorPage

Gregory-Vinokur committed:
Merge branch 'develop' into feat/add_adaptive_to_communities

artemkamyshenkov committed:
Merge branch 'main' into develop

Gregory-Vinokur committed:
Merge pull request #47 from Gregory-Vinokur/feat/add_adaptive_to_communities

feat: add adaptive and translate to communitiesPage, loginPage, newsP…

Gregory-Vinokur committed:
fix: fix getTimeDifference function

Gregory-Vinokur committed:
fix: fix Sport News generator

Gregory-Vinokur committed:
fix: change news API to newscatcher API for Netlify deploy

Gregory-Vinokur committed:
fix: fix infinity scroll bug
